### ISSUE TITLE
Server-side validation error in shipping address form does not appear, UI unresponsive

### ISSUE THREAD

**Issue by zetlen** [magento-research/pwa-studio#454]
## This issue is for the following packages:

[x] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-devdocs`
[ ] `upward-js`
[ ] `upward-spec`

## This issue is a:

[x] Bug
[ ] Feature suggestion
[ ] Documentation issue
[ ] Other (Please Specify)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             | 2.3.0-beta32     |

## Description

Upon submitting the shipping address form, if the API responds with a validation error, the UI does not display the error and does not allow the user to correct it.

## Steps to reproduce

Prerequisites: Venia dev or staging server running, sample data populated

1. Run the dev or staging server
2. Visit a SimpleProduct URL, e.g. `/carmina-earrings.html`
3. Click "Add to Cart"
4. Click "Checkout"
5. In the "Ship to" row, click "Click to edit"
6. Put an invalid value in the State field, e.g. **Texarkana**. Leave the rest of the form empty.
7. Click "Save".
8. Try to change any form input to reset form validity state.

## Expected result:

A validation error appears on the form regarding the invalid State value. When the user changes the input, the validation error disappears and/or the Save button enables. The user can correct the input.

## Actual result:

The Save button is disabled, no validation message appears, and the user cannot change form state.

## Possible solutions:

Something is wrong with input change detection.

**Comment by pcvonz** [comment]
Just added simple validation to check if a field is populated or not. Not sure how exactly we want the UX for this to go, but here is an example of what I have so far: 

![shipping-validation](https://user-images.githubusercontent.com/6378569/49035845-d65c4a00-f16a-11e8-83f0-2c382a18ba9b.gif)

**Comment by pcvonz** [comment]
Seems to be a duplicate of #462?

**Comment by zetlen** [comment]
Closing in favor of #462, which has more activity and data.
